fvm/evm/handler/blockstore.go

```diff
@@ -92,9 +100,56 @@ func (bs *BlockStore) LatestBlock() (*types.Block, error) {
 }
 
 // BlockHash returns the block hash for the last x blocks
-//
-// TODO: implement this properly to keep the last 256 block hashes
-// and connect use it inside the handler to pass as a config to the emulator
-func (bs *BlockStore) BlockHash(height int) (gethCommon.Hash, error) {
-	return gethCommon.Hash{}, nil
+func (bs *BlockStore) BlockHash(height uint64) (gethCommon.Hash, error) {
+	bhl, err := bs.getBlockHashList()
+	if err != nil {
+		return gethCommon.Hash{}, err
+	}
+	if bhl == nil {
+		if height == 0 {
+			return types.GenesisBlockHash, nil
+		}
+		return gethCommon.Hash{}, nil
```

Comment:
wouldn't this be an error case, where is normal for this to happen?

Reply:
Unfortunately that's the expected behaviour for Geth, if a height is not there it is not an error it returns an empty hash only